lib/src/socket_client.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:uuid/uuid.dart';
+
+import '../graphql_flutter.dart';
+
+SocketClient socketClient;
+
+class SocketClient {
+  static Future<SocketClient> connect(final String endPoint, {
+    final List<String> protocols = const ['graphql-ws'],
+    final Map<String, String> headers = const {'Content-Type': 'application/json'}
+  }) async {
+    return SocketClient(GraphQLSocket(await WebSocket.connect(
+        endPoint, protocols: protocols, headers: headers)));
+  }
+
+  final Uuid _uuid = new Uuid();
+  final GraphQLSocket _socket;
+
+  SocketClient(this._socket) {
+    _socket.connectionAck.listen(print);
+    _socket.connectionError.listen(print);
+    _socket.unknownData.listen(print);
+    _socket.write(InitOperation());
+  }
+
+  Stream<SubscriptionData> subscribe(final SubscriptionRequest payload) {
+    var id = _uuid.v4();
```

Comment:
Change var to a more specific type

Reply:
In this case I think I agree that `var` is better. The method `Uuid::v4` returns a dynamic type, and denotes it may not always be a string depending on the arguments provided. Perhaps it would be better to explicitly call `toString()` on it as well for safety... 

For other cases I'd like to debate the comment you made:  

> "give types to variables not just 'var' (you can just hover over the method or variable and it will you give the return type":

In the official [dart style guide](https://www.dartlang.org/guides/language/effective-dart/design#avoid-type-annotating-initialized-local-variables) they specifically say to prefer `var` for local scoped variables. Type inference works pretty well in the local scopes, and leaves the denser logic less cluttered. It seems like a waste of the language feature to not embrace it.

> Local variables, especially in modern code where functions tend to be small, have very little scope. Omitting the type focuses the reader’s attention on the more important name of the variable and its initialized value.

To that point I'm open to any styling rules as long as we can stay consistent throughout the project. I'll start looking to see if there are any good static analysis/ checkstyle tools we can apply to the build such that we can begin codifying these style issues!

Reply:
Ah, on my second search I see that flutter has their own style guidelines that contradicts the dart-lang crew: 

> Always avoid "var". Use "dynamic" if you are being explicit that the type is unknown. Use "Object" if you are being explicit that you want an object that implements == and hashCode.

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-var

Seems rather restrictive to me, but I guess staying consistent with the framework is more useful here?

¯\\_(ツ)_/¯

Reply:
Well they are quite contradicting, I think it's best to pick one for this repo and stick with it. Personally the [Flutter style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo) seems like a more appropriate solution for this project, because the project is a Flutter library. But feel free to change my mind.

Reply:
Yeah I agree with using the flutter spec for now! In the future if the project splits into two packages then perhaps we should follow dart-lang for the base package and flutter for the Widgets package.

Reply:
Yeah good idea